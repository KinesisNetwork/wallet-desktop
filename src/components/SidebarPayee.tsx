import { WalletView } from '@types'
import * as React from 'react'

export interface Props {
  changeWalletView: (view: WalletView) => any
}

export const SidebarPayee: React.SFC<Props> = ({ changeWalletView }) => (
  <div className='has-text-centered'>
    <button className='button is-outlined is-fullwidth' onClick={() => changeWalletView(WalletView.payees)}>
      Manage Payees
    </button>
  </div>
)

SidebarPayee.displayName = 'SidebarPayee'
