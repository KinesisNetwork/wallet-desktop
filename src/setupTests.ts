import { configure } from 'enzyme'
import * as Adapter from 'enzyme-adapter-react-16'

export default configure({ adapter: new Adapter() })
