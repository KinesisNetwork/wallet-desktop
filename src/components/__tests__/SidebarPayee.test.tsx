import { shallow } from 'enzyme'
import * as React from 'react'

import { SidebarPayee } from '@components/SidebarPayee'
import '../../setupTests'

describe('SidebarPayee', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<SidebarPayee />)

    expect(wrapper).toMatchSnapshot()
  })

  it('renders a button', () => {
    const wrapper = shallow(<SidebarPayee />)

    const button = wrapper.find('NavLink')

    expect(button).toHaveLength(1)
  })
})
