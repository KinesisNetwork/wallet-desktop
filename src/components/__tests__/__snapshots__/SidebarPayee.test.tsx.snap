// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarPayee renders correctly 1`] = `
<div
  className="has-text-centered"
>
  <NavLink
    activeClassName="is-active"
    aria-current="page"
    className="button is-outlined is-fullwidth"
    to="/address-book"
  >
    Manage Payees
  </NavLink>
</div>
`;
